packages/edit-site/src/components/sidebar-dataviews/dataview-item.js

```diff
@@ -29,14 +29,15 @@ export default function DataViewItem( {
 	suffix,
 } ) {
 	const {
-		params: { path },
+		params: { path, layout },
 	} = useLocation();
 
 	const iconToUse =
 		icon || VIEW_LAYOUTS.find( ( v ) => v.type === type ).icon;
 
 	const linkInfo = useLink( {
 		path,
+		layout,
```

Comment:
Why do we need this change? In pages list it certainly creates problems when changing views. For example if we change a view from the sidebar which has a `grid` view set, I'd expect this to be respected without taking into account the current `layout` param set.

In general the logic is hard to follow with the handling of views in sidebar and it would be great if we can simplify and also have more consistency between the different lists(pages, templates).